docs/source/tensorbay_cli/completion.rst

```diff
@@ -0,0 +1,180 @@
+other shells too.
+
+Shell completion suggests command names and option names. Options are only listed if at least a dash has been entered.
+Hidden commands and options are not shown.
+
+.. code:: console
+
+    $ gas <TAB><TAB>
+    auth     -- Authenticate the accessKey of gas.
+    branch   -- List, create or delete branches.
+    commit   -- Commit drafts.
+    config   -- Configure the options when using gas CLI.
+    cp       -- Copy local data to a remote path.
+    dataset  -- List, create or delete datasets.
+    draft    -- List or create drafts.
+    log      -- Show commit logs.
+    ls       -- List data under the path.
+    rm       -- Remove the remote data.
+    tag      -- List, create or delete tags.
+    $ gas auth -<TAB><TAB>
+    --get    -g  -- Get the accesskey of the profile
+    --unset  -u  -- Unset the accesskey of the profile
+    --all    -a  -- All the auth info
+    --help       -- Show this message and exit.
+
+Activation
+***********
+
+Completion is only available if a script is installed and invoked through an entry point, not through the ``python``
+command. Once the executable is installed, calling it with a special environment variable will put CLI in completion
+mode.
+
+In order for completion to be used, the user needs to register a special function with their shell. The script is
+different for every shell. The built-in shells are ``bash``, ``zsh``, and ``fish``. The following instructions will lead
+user to configure the completion:
+
+Before configuring completion, the user needs to check the version of ``click``:
+
+.. code:: console
+
+    $ pip show click
+    Name: click
+    Version: 8.0.1
+    Summary: Composable command line interface toolkit
+    Home-page: https://palletsprojects.com/p/click/
+    Author: Armin Ronacher
+    Author-email: armin.ronacher@active-4.com
+
+Activation for Click 7.x
+-------------------------
+
+For Bash:
+    Add this to ``~/.bashrc``:
+
+    .. code:: console
+
+        eval "$(_GAS_COMPLETE=source_bash gas)"
+
+For Zsh:
+    Add this to ``~/.zshrc``:
+
+    .. code:: console
+
+        eval "$(_GAS_COMPLETE=source_zsh gas)"
+
+For Fish:
+    Add this to ``~/.config/fish/completions/gas.fish``:
+
+    .. code:: console
+
+        eval (env _GAS_COMPLETE=source_fish gas)
+
+Activation for Click 8.x
+-------------------------
+
+For Bash:
+    Add this to ``~/.bashrc``:
+
+    .. code:: console
+
+        eval "$(_GAS_COMPLETE=bash_source gas)"
+
+For Zsh:
+    Add this to ``~/.zshrc``:
+
+    .. code:: console
+
+        eval "$(_GAS_COMPLETE=zsh_source gas)"
+
+For Fish:
+    Add this to ``~/.config/fish/completions/gas.fish``:
+
+    .. code:: console
+
+        eval (env _GAS_COMPLETE=fish_source gas)
+
+Activation Script
+******************
+
+Using ``eval`` means that the command is invoked and evaluated every time a shell is started, which can delay shell
+responsiveness. To speed it up, write the generated script to a file, then source that.
```

Comment:
This is a little ambiguous. Is it to use activation script instead of eval or to use activation script and eval?